src/systems/particle_emitter/ParticleEmitter.cc

```diff
@@ -71,10 +71,47 @@ void ParticleEmitter::Configure(const Entity &/*_entity*/,
     return;
   }
 
+  // allow_renaming
+  bool allowRenaming = false;
+  if (_sdf->HasElement("allow_renaming"))
+    allowRenaming = _sdf->Get<bool>("allow_renaming");
+
   // Name.
   std::string name = "particle_emitter_entity_" + std::to_string(entity);
   if (_sdf->HasElement("emitter_name"))
-    name = _sdf->Get<std::string>("emitter_name");
+  {
+    std::set<std::string> emitterNames;
+    std::string emitterName = _sdf->Get<std::string>("emitter_name");
+
+    // check to see if name is already taken
+    _ecm.Each<components::Name, components::ParticleEmitter>(
+        [&emitterNames](const Entity &, const components::Name *_name,
+                      const components::ParticleEmitter *)
+        {
+          emitterNames.insert(_name->Data());
+          return true;
+        });
+
+    name = emitterName;
+
+    // rename emitter if needed
+    if (emitterNames.find(emitterName) != emitterNames.end())
+    {
+      if (!allowRenaming)
+      {
+        ignwarn << "Entity named [" << name
+                << "] already exists and "
+                << "[allow_renaming] is false. Entity not spawned."
+                << std::endl;
+        return;
+      }
+      int counter = 0;
+      while (emitterNames.find(name) != emitterNames.end())
+      {
+        name = emitterName + "_" + std::to_string(++counter);
+      }
```

Comment:
Should we print a message after renaming that lets the user know what the emitter was renamed to?

Reply:
done. ff65cb6